src/library/scala/collection/MapView.scala

```diff
@@ -125,6 +127,21 @@ object MapView extends MapViewFactory {
     override def isEmpty: Boolean = iterator.isEmpty
   }
 
+  @SerialVersionUID(3L)
+  class TapEach[K, +V, +U](underlying: SomeMapOps[K, V], f: ((K, V)) => U) extends AbstractMapView[K, V] {
+    override def get(key: K): Option[V] = {
+      underlying.get(key) match {
+        case s @ Some(v) =>
+          f((key, v))
+          s
+        case None => None
+      }
+    }
+    override def iterator: Iterator[(K, V)] = underlying.iterator.tapEach(f)
+    override def knownSize: Int = if (underlying.knownSize == 0) 0 else super.knownSize
```

Comment:
It could just be `override def knownSize = underlying.knownSize`, no?

Reply:
@julienrf sorry, changed it to `underlying.knownSize`